Yoast/developer
# Various SEO fixes

## Summary
This PR applies various suggestions from @klaracvitko to improve the SEO of https://developer.yoast.com, based on [this internal document](https://docs.google.com/spreadsheets/d/1xfFuB7z9ALl3vH3E2aBNw-kkY7PcLjnpClPBAfOvCdI/edit?gid=0#gid=0). 

We did the following:
- prevented duplicate meta descriptions by removing overview pages and disambiguating some duplicate ones
- added meta descriptions for some category pages
- removed broken external links
- shortened some titles
- improved uninformative links

We also scouted the touched files for spelling errors. 

## Relevant technical choices
- The removed overview pages are redirected.
- When we introduced descriptions for some category pages, we redirected the original pages.

## Test instructions
This PR can be acceptance tested by following these steps:
1. Run the new report on https://fix-seo.yoast-developer.pages.dev/ and confirm the reported issues have been (mostly) resolved. 
2. Confirm the added redirects work.

## Quality assurance
